shopinvader/services/service.py

```diff
@@ -87,6 +87,9 @@ def _scope_to_domain(self, scope):
                     op = self._scope_to_domain_operators[op]
                 else:
                     op = "="
+                # If the key is a many2one field, we need to convert the value to an int
+                if key.endswith("_id") and isinstance(value, str) and value.isdigit():
```

Comment:
To make it stronger I will do something like
```suggestion
                if self.env[self._expose_model"]._fields[key].type in ["many2one", "many2many"] and isinstance(value, str) and value.isdigit():
```

Reply:
@paradoxxxzero, @hparfr